### PR TITLE
tls: add go-fuzz driver

### PR DESCRIPTION
**Pull request by daviddrysdale**
Instructions:
 go get -u github.com/dvyukov/go-fuzz/...
 go-fuzz-build github.com/google/certificate-transparency-go/tls/fuzz
 go-fuzz -bin=./fuzz-fuzz.zip -workdir=tls/fuzz/examples

### CODE REVIEW

**Comment by daviddrysdale** [comment]
This could (should?) live in a separate package to avoid cluttering up `tls` (cf. `x509util/fuzz.go`), but OTOH it's only one entrypoint and a small amount of code for users of the library to carry.  Thoughts?

File: tls/fuzz/fuzz.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+// OneOfEach has each basic type.
+type OneOfEach struct {
+	Val8   uint8
+	Val16  uint16
+	Val24  tls.Uint24
+	Val32  uint32
+	Val64  uint64
+	Choice tls.Enum `tls:"size:3"`
+	Buffer [12]byte
```
Comment (AlCutter) [inline tls/fuzz/fuzz.go]: Add a variable length field?

File: tls/fuzz/fuzz.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+type Something struct {
+	Inners []InnerType `tls:"minlen:1,maxlen:65535"`
+}
+
+// Fuzz is a go-fuzz (https://github.com/dvyukov/go-fuzz) entrypoint
+// for fuzzing the parsing of TLS-encoded data.
+func Fuzz(data []byte) int {
+	var result Outer
+	if _, err := tls.Unmarshal(data, &result); err == nil {
+		return 1 // Lexically correct
```
Comment (AlCutter) [inline tls/fuzz/fuzz.go]: Flip these around for "indent error" style?

File: tls/fuzz/fuzz.go (hunk 1 above)
Comment (daviddrysdale) [reply tls/fuzz/fuzz.go]: Variable length fields should already get hit by `InnerType`

File: tls/fuzz/fuzz.go (hunk 2 above)
Comment (daviddrysdale) [reply tls/fuzz/fuzz.go]: Done.

**Comment by daviddrysdale** [comment]
PTAL

File: tls/fuzz/fuzz.go (hunk 1 above)
Comment (AlCutter) [reply tls/fuzz/fuzz.go]: Yeah, but the comment says it has one of each basic type in here :)

**Pull request closed without merging**